lib/VM/JSLib/Intl.cpp

```diff
@@ -202,14 +209,24 @@ CallResult<std::vector<std::u16string>> normalizeLocales(
 
   ret.reserve(*lengthRes);
 
+  bool isProxy = localeObj->isProxyObject();
   if (LLVM_UNLIKELY(
           createListFromArrayLike(
               localeObj,
               runtime,
               *lengthRes,
-              [&ret](Runtime *runtime, uint64_t index, PseudoHandle<> value) {
-                CallResult<std::u16string> strRes =
-                    stringFromJS(runtime, std::move(value));
+              [&ret, isProxy](Runtime *runtime, uint64_t index, PseudoHandle<> value) {
+
+    // When the locales list is a proxy object, gaps are allowed.
```

Comment:
As far as I can tell, gaps are allowed for non-proxies, too.

Reply:
If I remember correctly, firefox throws range exception with normal arrays with gaps .. I couldn't get to the spec which covers it.

Reply:
I tried
`new Intl.Collator([, "en-US", , "fr-FR"]).resolvedOptions().locale`
on node, firefox, and jsc, and all returned "en-US" without throwing.  I believe the relevant spec is https://tc39.es/ecma402/#sec-canonicalizelocalelist step 7.b-c, which skips elements not present.

Reply:
new Intl.Collator([, "en-US", undefined , "fr-FR"]).resolvedOptions().locale is throwing a TypeError with chrome : { Language ID should be string or object. } .. but new Intl.Collator([, "en-US", , "fr-FR"]).resolvedOptions().locale is not .. 

From a developer perspective, both "['a',,'c'][1]" and "['a',undefined,'c'][1]" evaluates to undefined .. But in the locales array, a gap and "explicit undefined" doesn't mean the same !

Reply:
There's a difference between undefined and having a gap, though.  They do behave differently as described in the spec, but I don't see any expected difference in behavior for a Proxy vs a plain array.  In your example, `a[1]` are both undefined, but `Reflect.has(a, 1)` are not the same.